src/binary/ibc-setup/utils/load-and-validate-app.ts

```diff
@@ -0,0 +1,46 @@
+import fs, { appendFileSync } from 'fs';
+import os from 'os';
+import path from 'path';
+
+import Ajv, { JSONSchemaType } from 'ajv';
+import yaml from 'js-yaml';
+
+import { appFile } from '../../constants';
+import { App } from '../types';
+
+export function loadAndValidateApp(home: string) {
+  const appFilePath = path.join(home, appFile);
+
+  if (!fs.existsSync(appFilePath)) {
```

Comment:
Good point about race conditions.

Does it look better?
8ae637e

I'm thinking if we should try..catch the `yaml.load` (both in app and registry loading)? WYT?

Reply:
I think if `fs.readFileSync` throws we don't want to silently swallow that error. It probably means the user intended to use a config file but it doesn't have the right permissions or they gave us the enclosing folder instead of the file or something.

Reply:
The thing is that `app.yaml` is optional so it may not-exist. In such case we'd like to return `null` from `loadAndValidateApp`

Reply:
Yeah, but you'll get different errors for different situations. I think there are at least a couple where we want to exit immediately and tell the user they're doing something wrong.

Reply:
I think the original logic worked relatively well. It silently ignored missing app.yaml and threw errors if it was a directory or we didn't have read permissions.

There is a race condition, but that is a super edge case. Just make an issue for that. We are going for 0.1.0 release. Let's try to get "good enough" out and add issues for ways to improve it in the future

Reply:
I don't have a strong preference over either of the two approaches done. Both have limits. Let's mark it, merge and move forward. Merging this blocks writing more commands

Reply:
I pushed something that eliminates race condition and returns `null` if the file doesn't exist. It throws on other errors though such as no permissions.

a66467d